src/wallet/wallet.cpp

```diff
@@ -2587,6 +2589,17 @@ bool CWallet::CreateTransaction(const vector<CRecipient>& vecSend, CWalletTx& wt
     // all.
     txNew.nLockTime = std::max(0, chainActive.Height() - 10);
 
+    // Activates after Overwinter network upgrade
+    // Set nExpiryHeight to 20 blocks (default) past current blockheight
+    if (NetworkUpgradeActive(chainActive.Height(), Params().GetConsensus(), Consensus::UPGRADE_OVERWINTER)) {
+        txNew.nVersion = 3;
+        if (chainActive.Height() + blockExpiry > TX_EXPIRY_HEIGHT_THRESHOLD){
+            strFailReason = _("nExpiryHeight must be less than TX_EXPIRY_HEIGHT_THRESHOLD.");
+        } else {
+            txNew.nExpiryHeight = chainActive.Height() + blockExpiry;
+        }
```

Comment:
Per earlier comment, move this into `CreateNewContextualCMutableTransaction`.

Reply:
I don't think this logic should move into CreateNewContextualCMutableTransaction for reasons given in above comment.  No need to spread the use of globals like chainActive and blockExpiry any more than necessary.